java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -411,6 +413,85 @@ public static CompletableFuture<GlideString> get(
                 new ArgsBuilder().add(gs(JSON_GET)).add(key).add(options.toArgs()).add(paths).toArray());
     }
 
+    /**
+     * Retrieves the JSON values at the specified <code>path</code> stored at multiple <code>keys
+     * </code>.
+     *
+     * @apiNote When in cluster mode, if keys in <code>keys</code> map to different hash slots, the
+     *     command will be split across these slots and executed separately for each. This means the
+     *     command is atomic only at the slot level. If one or more slot-specific requests fail, the
+     *     entire call will return the first encountered error, even though some requests may have
+     *     succeeded while others did not. If this behavior impacts your application logic, consider
+     *     splitting the request into sub-requests per slot to ensure atomicity.
+     * @param client The client to execute the command.
+     * @param keys The keys of the JSON documents.
+     * @param path The path within the JSON documents.
+     * @return An array with requested values for each key.
+     *     <ul>
+     *       <li>For JSONPath (path starts with <code>$</code>): Returns a stringified JSON list
+     *           replies for every possible path, or a string representation of an empty array, if
+     *           path doesn't exist.
+     *       <li>For legacy path (path doesn't start with <code>$</code>): Returns a string
+     *           representation of the value in <code>path</code>. If <code>path</code> doesn't exist,
+     *           the corresponding array element will be <code>null</code>.
+     *     </ul>
+     *     If a <code>key</code> doesn't exist, the corresponding array element will be <code>null
+     *     </code>.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc1", "$", "{\"a\": 1, \"b\": [\"one\", \"two\"]}").get();
+     * Json.set(client, "doc2", "$", "{\"a\": 1, \"c\": false}").get();
+     * var res = Json.mget(client, new String[] { "doc1", "doc2", "doc3" }, "$.c").get();
+     * assert Arrays.equals(res, new String[] { "[]", "[false]", null });
+     * }</pre>
+     */
+    public static CompletableFuture<String[]> mget(
+            @NonNull BaseClient client, @NonNull String[] keys, @NonNull String path) {
+        return Json.<Object[]>executeCommand(
+                        client, concatenateArrays(new String[] {JSON_MGET}, keys, new String[] {path}))
+                .thenApply(res -> castArray(res, String.class));
```

Comment:
A bit of confusion here. So 3 things:
1. the `thenApply()` call here on line 452
2. the `<Object[]>` type hint before invoking `executeCommand()` on line 450
3. the final convertion in `executeCommand()`, which will convert to whatever the method return type is (`.thenApply(r -> (T) r)`) on line 2709 (for `mget()`, `T` is `String[]`)

Not sure why all 3 are needed here. Isn't `3` alone good enough?

Reply:
`lib.rs` returns an arbitrary `Object[]`, it cannot be casted to `String[]`, so we have to use `castArray`.
So what happens here:
1. libglide returns `Object[]`
2. `customCommand` returns `Object` (which is actually array, yes)
3. with `Json.<Object[]>` prefix we instruct that we expect `Object[]`
4. with `.thenApply(r -> (T) r)` we safely cast this `Object` to `Object[]`
5. with `castArray` we convert `Object[]` to `String[]`

So in total there is a chained `Object` -> `Object[]` -> `String[]` conversion distrubuted over multiple places of code 🤷 
I'd like to refactor this if you give me an idea.

Reply:
Also note:  we kind of have to go through Object -> Object[] -> String[]. 
This is because castArray requires the Object[] type.  We cannot cast Object[] to String[] directly, but we need to cast all inner elements (thank you Java).